src/Haskell/Ide/Engine/Plugin/ApplyRefact.hs

```diff
@@ -307,7 +307,10 @@ codeActionProvider plId docId _ context = IdeResultOk <$> hlintActions
   where
 
     hlintActions :: IdeM [LSP.CodeAction]
-    hlintActions = catMaybes <$> mapM mkHlintAction (filter validCommand diags)
+    hlintActions = do
+      actions <- mapM mkHlintAction (filter validCommand diags)
+      applyAll <- mkApplyAllAction
+      return (catMaybes $ applyAll:actions)
```

Comment:
Some clients like coc.vim have keybindings to apply the first code action, so would it make sense to return this last?
```suggestion
      return (catMaybes $ actions ++ [applyAll])
```